### PR TITLE
added scaffolding-chef plan

### PR DESCRIPTION
**Pull request by echohack**
After some heated discussion, we've determined that the best location for the Effortless pattern, and the scaffolding should belong in `chef/chef`.

scaffolding-chef is the framework and implementation of all of the best practices that people who use the Effortless pattern to run Chef Infra. We have developed this in collaboration with the community, and will continue to do so. 

This is primarily for two reasons:

1. scaffolding-chef is open source and developed in collaboration with the community. That means the Effortless pattern is also open source. Anyone can build any content they want on top of this pattern, and it's up to them to decide whether that content is open source or not. If someone wants to open source their content, they can contribute to the [public Chef Supermarket](https://supermarket.chef.io).

2. scaffolding-chef depends on packages in the `chef` origin, which is gated by the expeditor builds that come from this repository, from the /habitat/ directory in the root of this project. It only makes sense that any package that deps on packages from the chef origin should also belong in this project.

Using core-plans as a model, I've put the scaffolding in the root of the project.

Additionally, this adds a brand new plan.ps1 and scaffolding.ps1 to enable others to create windows based scaffolding-chef packages.

During this process we discovered that Windows plan-build.ps1 does not honor scaffolding dependencies in quite the same way as the Linux plan-build.sh scaffolding dependency code. I'll be opening an issue about it soon. Until then, there is a workaround available by simply including this in your consuming package:

```
$pkg_deps=@(
  "stuartpreston/chef-client-detox"
  "core/cacerts"
)
$pkg_build_deps=@("core/chef-dk", "core/git")
```

One final note, this windows package is currently dependent on some offshoot, custom built versions of chef-client for Windows. These need to be fixed upstream before switching the origins over to `chef/chef-client`. Work is ongoing to get this done -- and this PR is not tightly coupled to that work. Consumers of the scaffolding can continue to work with the offshoot builds with no future disruption in how they build their cookbooks.

### Testing

Create a consuming plan.ps1 that looks like the following

```
$pkg_name="effortless"
$pkg_origin="example"
$pkg_version="0.1.0"
$pkg_deps=@(
  "stuartpreston/chef-client-detox"
  "core/cacerts"
)
$pkg_scaffolding="echohack/scaffolding-chef"
$scaffold_policy_name="windows"
$scaffold_policyfile_path="$PLAN_CONTEXT"
```

You will also need a policyfile in the same directory as the plan.ps1.

```
name "windows"

default_source :supermarket
default_source :chef_repo, './cookbooks' do |s|
  s.preferred_for 'windows'
end

run_list ['hardening::default']
```

You will also need some cookbooks. You can grab these over at: https://github.com/echohack/core-plans/pull/2/files

I have been building the Windows scaffolding on my origin, so by pointing to my origin, you'll get the latest scaffolding. You can always build it yourself and point at your own origin too.

breathtaking

### CODE REVIEW

**Comment by TheLunaticScripter** [comment]
This is the right move for the scaffolding and for core plans. I'm excited about this change.

**Comment by echohack** [comment]
The build failures are completely unrelated to this change. Can someone override?

```
1) chef-client when the chef repo has a cookbook with a no-op recipe should complete with success when using --profile-ruby and output a profile file
     Failure/Error: cmd.error!
     Mixlib::ShellOut::ShellCommandFailed:
       Expected process to exit with [0], but received '1'
       ---- Begin output of bundle exec chef-client --minimal-ohai -c "C:/Users/appveyor/AppData/Local/Temp/1/chef_repo20190613-4284-g2zmse/config/client.rb" -o 'x::default' -z --profile-ruby ----
       STDOUT: [2019-06-13T19:04:05+00:00] FATAL: RuntimeError: You must have the ruby-prof gem installed in order to use --profile-ruby
       STDERR: 
       ---- End output of bundle exec chef-client --minimal-ohai -c "C:/Users/appveyor/AppData/Local/Temp/1/chef_repo20190613-4284-g2zmse/config/client.rb" -o 'x::default' -z --profile-ruby ----
       Ran bundle exec chef-client --minimal-ohai -c "C:/Users/appveyor/AppData/Local/Temp/1/chef_repo20190613-4284-g2zmse/config/client.rb" -o 'x::default' -z --profile-ruby returned 1
     # ./vendor/bundle/ruby/2.5.0/gems/mixlib-shellout-3.0.4-universal-mingw32/lib/mixlib/shellout.rb:297:in `invalid!'
     # ./vendor/bundle/ruby/2.5.0/gems/mixlib-shellout-3.0.4-universal-mingw32/lib/mixlib/shellout.rb:284:in `error!'
     # ./lib/chef/mixin/shell_out.rb:132:in `shell_out_compacted!'
     # ./lib/chef/mixin/shell_out.rb:66:in `shell_out!'
     # ./spec/integration/client/client_spec.rb:358:in `block (3 levels) in <top (required)>'
```

```

2) knife data bag from file When the Chef server has an empty data bag when the chef repo has some data bag items uploads a file in chef's internal format
```

**Comment by tas50** [comment]
@echohack do we really have to stick this in the root of the directory here. Can this just go in the hab dir we already have?

**Comment by TheLunaticScripter** [comment]
So one of the reasons that I think this needs to be in its own directory instead of in the habitat directory is to separate the habitat build of Chef Infra from the recommended approach for running it with the Effortless Config model. I also think this needs a `.bldr.toml` so that both the scaffolding and the client can get built properly with expeditor.

**Comment by echohack** [comment]
@tas50 The `habitat` directory in the root is for building the `chef/chef-client` plan. It's maybe best to move both to a `habitat-packages` directory, and rename `habitat` to `chef-client`.

What do you think about that?

**Comment by echohack** [comment]
Build is still drunk and will need to be overriden to merge.

**Comment by echohack** [comment]
Build is still dead. 

Travis Build:Rubocop is dying on... not ruby handlebars syntax.

AppVeyor build is dying on Chef Server knife data bag upload, which I'm 100% positive a Habitat plan has nothing to do with. ^_^

File: habitat-packages/scaffolding-chef/plan.ps1 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+$pkg_name="scaffolding-chef"
+$pkg_description="Scaffolding for Chef Policyfiles"
+$pkg_origin="chef"
+$pkg_version="0.6.0"
+$pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
```
Comment (tas50) [inline habitat-packages/scaffolding-chef/plan.ps1]: this is the chef maintainers now

File: habitat-packages/scaffolding-chef/plan.ps1 [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+$pkg_license=("Apache-2.0")
+$pkg_upstream_url="https://www.chef.sh"
+$pkg_build_deps=@(
+    "core/chefdk"
```
Comment (tas50) [inline habitat-packages/scaffolding-chef/plan.ps1]: we should get this on workstation at some point

**Pull request merged**